nexus/db-model/src/dataset.rs

```diff
@@ -4,11 +4,14 @@
 
 use super::{DatasetKind, Generation, Region, SqlU16};
 use crate::collection::DatastoreCollectionConfig;
+use crate::ipv6;
 use crate::schema::{dataset, region};
 use chrono::{DateTime, Utc};
 use db_macros::Asset;
+use internal_dns_client::names::{BackendName, ServiceName, AAAA, SRV};
+use nexus_types::identity::Asset;
 use serde::{Deserialize, Serialize};
-use std::net::SocketAddr;
+use std::net::{Ipv6Addr, SocketAddrV6};
 use uuid::Uuid;
 
 /// Database representation of a Dataset.
```

Comment:
Cool.  The changes here make sense.

This is unrelated to this PR but I find it a little unintuitive that we call this a `Dataset` because it seems to encapsulate not just the ZFS dataset, but at least some aspects of the networking for the service running in a zone atop the dataset (its IP, its DNS record type and record name).  Is that just historical or is there some intuition I'm missing?

Reply:
That's a fair point, it's basically tightly coupling the concept of the "dataset" with "the service which manages the dataset".

This has been implemented because it fits our current "dataset-managing services", including:
- CRDB
- Crucible
- Clickhouse

Though it's a fair point, it definitely won't apply to all "datasets", and it makes it awkward to describe datasets which aren't managed by an explicit service (such as a dump device, dataset for logs, etc).

Filed https://github.com/oxidecomputer/omicron/issues/2000